src/main/java/eu/europeana/metis/sandbox/domain/RecordProcessEvent.java

```diff
@@ -0,0 +1,63 @@
+package eu.europeana.metis.sandbox.domain;
+
+import eu.europeana.metis.sandbox.common.OaiHarvestData;
+import eu.europeana.metis.sandbox.common.Status;
+import eu.europeana.metis.sandbox.common.Step;
+import java.util.List;
+
+/**
+ * Event that contains a record and its processing details
+ */
+public class RecordProcessEvent {
```

Comment:
While this was an event that applied to every step, it is now polluted with information of oai harvesting specifically that do not apply to other steps.
I think we should decouple this somehow.

Reply:
I completely agree with you, there should be different type of events containing different type of information.
I'm not sure if this is considered to be part of the ticket, as in if we should focus on this, but we could create a separate ticket dedicated to the improvement of the queue messaging. But I guess this depends on the outcome of the discussion brought up in `AsyncDatasetPublishService`.
What do you think?

Reply:
Indeed, probably fixing the asynchronous issue would not require this information here at all.